be/dtm0_log.h

```diff
@@ -0,0 +1,236 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2013-2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#pragma once
+#ifndef __MOTR_BE_DTM0_LOG_H__
+#define __MOTR_BE_DTM0_LOG_H__
+
+/**
+ *  @page dtm0 log implementation
+ *
+ *  @section Overview
+ *  DTM0 log module will be working on incoming message request, the
+ *  goal of this module is to track journaling of incoming message either on
+ *  persistent or volatile memory based on whether logging is happening on
+ *  participant or on originator, so that in the phase failure consistency of
+ *  failed participant can be restored by iterating over logged journal to
+ *  decide which of the logged messages needs to be sent as a redo request.
+ *
+ *  During normal operation when every participant and originator is online
+ *  every transaction will be logged and at any point state of transaction can
+ *  be any of the following,
+ *
+ *	enum m0_dtm0_log_participant_state {
+ *		M0_DTML_STATE_UNKNOWN,
+ *		M0_DTML_STATE_SENT,
+ *		M0_DTML_STATE_EXECUTED,
+ *		M0_DTML_STATE_PERSISTENT,
+ *	};
+ *
+ *  Every participant maintains the journal record corresponds to each
```

Comment:
Suggestion: s/the journal record corresponds/a journal record that corresponds.

Rationale: a subordinating conjunction is missing between two the independent clause and the dependent clause.

Reply:
Addressed at 4e0d3688b2cc72748d9c1ed8df1ddeb8a2319fba